### ISSUE TITLE
tools/c7n-org - integer variable can't be deployed.

### ISSUE THREAD

**Issue by gnokoheat** [cloud-custodian/cloud-custodian#7879]
### Describe the bug

integer variable can't be deployed.

config file
```
accounts:
- account_id: xxxxxxxxxxxxxx
  vars:
    log-group-retention-in-days: 30
```
policy file
```
actions:
- type: retention
  days: {log-group-retention-in-days}
```
deploy result : lambda - config.json
```
"actions": [
  {
    "type": "retention",
    "days": {
      "log-group-retention-in-days": null
    }
  }
]
```

So, formatting variable with quote.

policy file
```
actions:
- type: retention
  days: "{log-group-retention-in-days}"
```
deploy result : lambda - config.json
```
"actions": [
  {
    "type": "retention",
    "days": "30"
  }
]
```
action result : error
```
{
  "errorMessage": "Parameter validation failed:\nInvalid type for parameter retentionInDays, value: 30, type: <class 'str'>, valid types: <class 'int'>",
  "errorType": "ParamValidationError",
  ...
}
```

### What did you expect to happen?

c7n-org integer variable can't be deployed. If changed it as string, c7n lambda throws an error.

### Cloud Provider

Amazon Web Services (AWS)

### Cloud Custodian version and dependency information

```shell
Custodian:   0.9.18
Python:      3.9.6 (v3.9.6:db3ff76da1, Jun 28 2021, 11:49:53)
```

**Comment by ajkerrigan** [comment]
This appears to be a duplicate of #7553, recently addressed in #7832 which allows variable types to pass through.

**Comment by kapilt** [comment]
closing as resolved in #7832.
